src/TransloaditClient.js

```diff
@@ -691,25 +694,54 @@ class TransloaditClient {
     })
 
     if (method === 'post' || method === 'put' || method === 'del') {
-      const extraData = Object.assign(
-        {tus_num_expected_upload_files: opts.tus_num_expected_upload_files},
-        opts.fields
-      )
+      const extraData = Object.assign({}, opts.fields)
+      if (opts.tus_num_expected_upload_files) {
+        extraData.tus_num_expected_upload_files = opts.tus_num_expected_upload_files
+      }
       this._appendForm(req, opts.params, extraData)
     }
   }
 
+  // @todo support size retrieval for other streams
+  _canGetStreamSizes () {
+    for (const label in this._streams) {
+      const stream = this._streams[label]
+      // the request module has path attribute that is different from file path
+      // but it also has the attribute httpModule
+      if (!(stream.path && !stream.httpModule)) {
+        return false
+      }
+    }
+
+    return true
+  }
+
   _sendTusRequest (opts, cb, onProgress) {
     let uploadsDone = 0
     const streamLabels = Object.keys(this._tus_streams)
     const tlClient = this
     let totalBytes = 0
-    let uploadedBytes = 0
+    let lastEmittedProgress = 0
+    let uploadProgresses = {}
     onProgress = onProgress || (() => {})
     for (const label of streamLabels) {
       const file = this._tus_streams[label]
-      const uploadSize = fs.statSync(file.path).size // todo this will fail for streams without path
+      const uploadSize = fs.statSync(file.path).size
```

Comment:
Very nice work, Ife :+1: Would you mind making this stat(1) call asynchronous with proper error handling? That would be the last remark I have